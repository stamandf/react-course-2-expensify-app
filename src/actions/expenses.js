import uuid from 'uuid';

// ADD_EXPENSE Action
export const addExpense = ({description = '', note = '', amount = 0, createdAt = 0 } = {}) => ({
    type: 'ADD_EXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
});

export const removeExpense = ({id} = {}) => ({
    type: 'DELETE_EXPENSE',
    id
});
// EDIT_EXPENSE
export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
});
