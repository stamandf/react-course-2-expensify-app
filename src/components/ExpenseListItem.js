import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    
        <li><Link to={"/edit/"+id}>{description}</Link>: {numeral(amount/100).format('$0,0.00')} - {moment(createdAt).format('MMMM Do, YYYY')}</li>
    )

export default ExpenseListItem;