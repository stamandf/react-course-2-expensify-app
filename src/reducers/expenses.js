// Expenses reducer
const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {  //Action creator
    switch (action.type) {
        case 'ADD_EXPENSE':
            // return state.concat(action.expense);
            return [...state, action.expense]; //The spreading operator for arrays
        case 'DELETE_EXPENSE':
            return state.filter(({id}) => id != action.id);  
        case 'EDIT_EXPENSE':   
            return state.map((expense) => {
                if(expense.id === action.id) {
                    return {
                        ...expense,
                        ...action.updates
                    };
                } else {
                    return expense;
                };
            });
        default:
            return state;
    }
};
