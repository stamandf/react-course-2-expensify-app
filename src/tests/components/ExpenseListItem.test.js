import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';


test('should render an expense item correctly', () => {
  const wrapper = shallow(<ExpenseListItem item={expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
});