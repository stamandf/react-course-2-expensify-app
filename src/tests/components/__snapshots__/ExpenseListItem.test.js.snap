// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an expense item correctly 1`] = `
<li>
  <Link
    replace={false}
    to="/edit/undefined"
  />
  : 
  $0.00
   - 
  December 31st, 1969
</li>
`;
